pkg/dns/vips/virtual_outbound.go

```diff
@@ -0,0 +1,143 @@
+package vips
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// VirtualOutbound the description of a hostname --> address and later port/tagSet
+type VirtualOutbound struct {
+	// This is not default in the legacy case (hostnames won't be complete)
+	Address   string         `json:"address,omitempty"`
+	Outbounds []MeshOutbound `json:"outbounds,omitempty"`
+}
+
+func (vo *VirtualOutbound) Equal(other *VirtualOutbound) bool {
+	if vo.Address != other.Address || len(vo.Outbounds) != len(other.Outbounds) {
+		return false
+	}
+	for i := range vo.Outbounds {
+		if vo.Outbounds[i].String() != other.Outbounds[i].String() {
+			return false
+		}
+	}
+	return true
+}
+
+type MeshOutbound struct {
+	Port   uint32
+	TagSet map[string]string
+	// A string to identify where this outbound was defined (usually the name of the outbound policy)
+	Origin string
+}
+
+func (mo *MeshOutbound) Less(o *MeshOutbound) bool {
+	return mo.Port < o.Port
+}
+
+func (mo *MeshOutbound) String() string {
+	var tags []string
+	for k, v := range mo.TagSet {
+		tags = append(tags, fmt.Sprintf("%s=%s", k, v))
+	}
+	sort.Strings(tags)
+	return fmt.Sprintf("%s=%d{%s}", mo.Origin, mo.Port, strings.Join(tags, ","))
+}
+
+type VirtualOutboundView struct {
```

Comment:
If this is opposed to GlobalView, I'd use `mesh` somewhere in the name. What do you think?